frontend/apps/cloudstorage/dropboxapi.lua

```diff
@@ -162,4 +170,52 @@ function DropBoxApi:showFiles(path, token)
     return dropbox_files
 end
 
+
+function DropBoxApi:fetchAdditionalFolders(response, token)
+  local out = response
+  local cursor = response.cursor
+
+  repeat
+    local num_entries = #out.entries
+    local data = "{\"cursor\": \"" .. cursor .. "\"}"
+
+    local sink = {}
+    socketutil:set_timeout()
+    local request = {
+        url     = API_LIST_ADD_FOLDER,
+        method  = "POST",
+        headers = {
+            ["Authorization"]  = "Bearer ".. token,
+            ["Content-Type"]   = "application/json",
+            ["Content-Length"] = #data,
+        },
+        source  = ltn12.source.string(data),
+        sink    = ltn12.sink.table(sink),
+    }
+    local headers_request = socket.skip(1, http.request(request))
+    socketutil:reset_timeout()
+    if headers_request == nil then
+        return nil
+    end
+
+    local result_response = table.concat(sink)
+    local ret, result = pcall(JSON.decode, result_response)
+
+    if not ret then
+      return nil
+    end
+    
+    for k,v in pairs(result.entries) do
+      out.entries[num_entries + k] = v
+    end
```

Comment:
I assume this is a pure array, and that nothing is ever going to be sparse, so:

```suggestion
    for __, v in ipairs(result.entries) do
      table.insert(out.entries, v)
    end
```

Reply:
(Untested, I don't use Dropbox ;)).

Reply:
(Luacheck will then complain that `num_entries` is unused, so drop it ;))